src/util/Concepts.hpp

```diff
@@ -46,4 +48,25 @@ hasNoDuplicates(auto&&... values)
     return (std::unique(std::begin(store), end) == end);
 }
 
+/**
+ * @brief Checks that the list of given type contains no duplicates
+ *
+ * @tparam Types The types to check
+ * @returns true if no duplicates exist; false otherwise
+ */
+template <typename... Types>
+constexpr bool
+hasNoDuplicateNames()
+{
+    constexpr std::array<std::string_view, sizeof...(Types)> names = {Types::name...};
+    for (std::size_t i = 0; i < names.size(); ++i) {
+        for (std::size_t j = i + 1; j < names.size(); ++j) {
+            if (names[i] == names[j]) {
```

Comment:
I think the Types don't need to be sorted to find duplicate, right?

Reply:
As I understand you go through an array of string_views and check each pair for equality. It will work if the array is sorted `{ Type_A, Type_A, Type_B }` but if it is not sorted it will not work, for example for `{ Type_A, Type_B, Type_A }` this function will not find duplicate.

Reply:
Sorry, my bad, the code is correct.